enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,264 @@
+  fields.
+
+### Variations
+
+- The operator will default to enabling Shipwright Builds, deploying its associated controllers and
+  custom resources. Cluster admins can disable Shipwright Builds, which removes the controller
+  deployment. Components related to the custom resources (CRDs, conversion webhook) are not
+  removed.
+- If the operator is deployed on a cluster with the `CSIDriverSharedResource` feature gate enabled,
+  changing `spec.sharedResources.enabled` will not be respected, and an appropriate status
+  condition will report why the operator has not deployed the Shared Resource CSI driver.
+
+
+### API Extensions
+
+The OpenShift Builds will intoroduce a new configuration CRD, `OpenShiftBuild`. It will contain 
+`spec` fields that configure the Shared Resource CSI driver in addition to Shipwright-related
+components. The operator will also deploy (directly or indirectly) the following:
+
+- Custom resources for the Shipwright Operator
+- Custom resources and webhooks for Shipwright Build
+- Custom resources and webhooks for the Shared Resource CSI Driver
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+As noted above, the operator will not deploy the Shared Resource CSI Driver if the cluster has
+the respective OCP feature gate enabled. This generally applies to tech preview clusters or
+clusters with custom fature gate tunings.
+
+With the start of OCP 4.16, the Shared Resource CSI driver operator will be removed from the
+[Cluster Storage Operator](https://github.com/openshift/cluster-storage-operator). The feature
+will likewise be removed from OpenShift's feature sets - any code that needs the most up to date
+set of feature sets will need to be updated (ex - openshift/library-go).
+
+The Shared Resource CSI driver itself will need to be removed from the OpenShift payload. Its
+components will need to be re-productized as an OLM operand.
+
+
+### Risks and Mitigations
+
+**Risk**: Same CSI driver will be deployed twice on tech preview clusters.
+*Mitigation*: Operator will check if the `CSIDriverSharedResource` feature is enabled before
+deploying the Shared Resource CSI driver itself.
+
+
+**Risk**: Admins or users will want the Shared Resource CSI Driver without Shipwright
+*Mitigation*: Shipwright components can be disabled via the `OpenshiftBuild` CRD.
+
+
+### Drawbacks
+
+- Shared resources no longer a part of OpenShift
+- Customers who need entitlements in builds will need to install a new operator, or resort to less
+  desirable work-arounds
+- Potential impact on Red Hat components that need shared resources and/or RHEL entitlements
+
+
+## Design Details
+
+
+### Test Plan
+
+- Shared resource logic in the OCP Builds tech preview suite will need to be migrated to the
+  OpenShift Builds operator test suites
+- OpenShift Builds operator will need a techpreview variant to ensure the shared resource CSI
+  driver is not managed by the operator on OCP 4.15 and earlier (tech preview).
```

Comment:
Can you add a line about moving (or removing) the e2e tests in openshift/origin that depend on the shared resource driver?
<https://github.com/openshift/origin/blob/master/test/extended/builds/volumes.go>
<https://github.com/openshift/origin/blob/master/test/extended/storage/inline.go>

Reply:
Updated